src/main/java/org/opensearch/security/securityconf/ConfigModelV6.java

```diff
@@ -829,12 +829,12 @@ private Set<String> getResolvedIndexPattern(User user, IndexNameExpressionResolv
                     .toArray(String[]::new);
 
                 if (aliasesForPermittedPattern.length > 0) {
-                    resolved = resolver.concreteIndexNames(cs.state(), IndicesOptions.lenientExpandOpen(), aliasesForPermittedPattern);
+                    resolved = resolver.concreteIndexNames(cs.state(), IndicesOptions.lenientExpand(), aliasesForPermittedPattern);
```

Comment:
Don't merge this PR yet, I am planning to push a commit to pass a boolean param in this method to `includeClosed` with default value of false. I plan to pass this param as `true` when computing the EvaluatedDlsFlsConfig.

Reply:
Pushed a change to pass `includeClosed` as a param to `ConfigV7.getResolvedIndexPattern` so that the caller determine how they want the resolution to proceed.

It makes sense to include Closed when computing the EvaluatedFlsDlsConfig, but should not break other use-cases.

For instance if you have 3 indices:

- **logs-1** - open
- **logs-2** - open
- **logs-3** - closed

When you search on `logs-*` the current behavior (need to double check) will only resolve to `logs-1` and `logs-2` when ultimately performing the search so `logs-3` is excluded. If this expansion mode is changed, then it includes `logs-3` in the resolution which it currently does not. I will write an integ test to confirm this search scenario.

Reply:
Its a bit more complicated then this. The security plugin has a setting to preserve the indices options from the original request, but it is disabled by default. See this section of [IndexResolverReplacer](https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/resolver/IndexResolverReplacer.java#L810-L823).

There's a few places in the security plugin that resolve patterns -> concrete indices:

1. On the original request, if the request contains index patterns then these will get resolved to concrete indices for authorization
2. When looping through roles, it will extract all patterns associated with index_permissions and resolve these to concrete indices

I think the logic I added into this PR is the right way to go. Consider the closed indices when computing the EvaluatedFlsDlsConfig, but keep the existing `IndicesOptions.lenientExpandOpen()` otherwise.

Reply:
> When you search on logs-* the current behavior (need to double check) will only resolve to logs-1 and logs-2 when ultimately performing the search so logs-3 is excluded. If this expansion mode is changed, then it includes logs-3 in the resolution which it currently does not. I will write an integ test to confirm this search scenario.

Thanks, good catch! This is an important scenario to test, DLS/FLS should not run search on closed indices.

Reply:
The security plugin does index pattern -> concrete resolution in 2 places.

1. When expanding out the `index_permissions` from roles definitions. It expands to concrete indices here to determine DLS/FLS/Field Masking rules per concrete index. See [ConfigV7.getDlsFls](https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/securityconf/ConfigModelV7.java#L276-L378)
2. When determining the indices from a request.

For the second use-case, I actually found a new dynamic setting from the security plugin that I wasn't aware of yet. `config.dynamic.respect_request_indices_options` which has a default value of `false`. If this value is set to `true` it will use the original requests rules when expanding concrete indices. For example, if a search request includes `expand_wildcards=all` then security would copy this. The default value for security is [here](https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/resolver/IndexResolverReplacer.java#L813).

```
IndicesOptions.fromOptions(false, true, true, false, true);
```

For reference, this is the default for a [SearchRequest](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/action/search/SearchRequest.java#L116)

See the full method here for determining IndicesOptions: https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/resolver/IndexResolverReplacer.java#L810-L823